nservicebus/messaging/messages-events-commands.md

```diff
@@ -50,6 +50,8 @@ Messages should:
 
 Note: Prior to version 7.2, messages had to be defined as a `class`. Defing them as a `struct` would result in a runtime exception.
 
+Generic message definitions (e.g. `MyMessage<T>`) are not supported. It is recommended to use dedicated, simple types for each message or to use inheritance to reuse shared message characteristics.
```

Comment:
Why so?

Reply:
@sungam3r we never implemented this as a fully supported feature and therefore don't have any automated tests to guarantee this to be working across all transports and other platform components, that's why do not recommend using generic message types.

If this is a feature you're strongly interested in, we'd be happy to learn more about your use case to review our current stance on this topic.

Reply:
That is, you want to say that they are not recommended to use, because you just have no enough tests?

Reply:
> That is, you want to say that they are not recommended to use, because you just have no enough tests?

because we never considered it a supported feature and therefore also don't have a reliable set of tests to confirm whether it would be working or not. There is some indication that it "works by accident" in certain environments but that's not an intentional feature. Therefore we also do not guarantee that this continues to work across different versions of NServiceBus for example.

Reply:
@sungam3r Can you perhaps explain what your use case is for wanting a generic message handler? We'd love to explore what you're trying to achieve.